openff/system/misc.py

```diff
@@ -0,0 +1,38 @@
+import mbuild as mb
+from openff.toolkit.topology import Molecule, Topology
+from simtk import unit
+
+
+def offmol_to_compound(off_mol: Molecule) -> mb.Compound:
+
+    if not off_mol.has_unique_atom_names:
+        off_mol.generate_unique_atom_names()
+
+    if off_mol.n_conformers == 0:
+        off_mol.generate_conformers(n_conformers=1)
+
+    comp = mb.Compound()
+    comp.name = off_mol.name
+
+    for a in off_mol.atoms:
+        atom_comp = mb.Particle(name=a.element.symbol)
+        comp.add(atom_comp, label=a.name)
+
+    for b in off_mol.bonds:
+        comp.add_bond((comp[b.atom1_index], comp[b.atom2_index]))
+
+    comp.xyz = off_mol.conformers[0].value_in_unit(unit.nanometer)
+
+    return comp
+
+
+def offtop_to_compound(off_top: Topology) -> mb.Compound:
+
+    sub_comps = []
+
+    for top_mol in off_top.topology_molecules:
+        this_comp = offmol_to_compound(top_mol.reference_molecule)
```

Comment:
This is a tiny bit risky because the "reference molecule" may have a different atom ordering than the "topology molecule" (this allows an OFF Topology to have multiple copies of the same molecule, with different atom orderings). This isn't a common situation, so I'm fine to move forward with the code in its current form for now. 
```suggestion
        # TODO: This could have unintended consequences if the TopologyMolecule has atoms in a different order than the reference Molecule
        this_comp = offmol_to_compound(top_mol.reference_molecule)
```

Reply:
Ah, I was unaware of that behavior. Thanks!